lib/logstash/inputs/stdin.rb

```diff
@@ -31,16 +36,21 @@ def run(queue)
         end
       rescue IOError, EOFError, LogStash::ShutdownSignal
         # stdin closed or a requested shutdown
+        @stop_requested.make_true
         break
+      rescue => e
+        # ignore any exception in the shutdown process
+        break if @stop_requested.true?
+        raise(e)
```

Comment:
the thing is that the current  path in the pipeline is really not clean so this is a safety. the real refactor will be correctly done in elastic/logstash#3211 and #1